app/src/main/java/com/example/inharmony/ChatAdapter.java

```diff
@@ -0,0 +1,204 @@
+            } else {
+                body.setVisibility(View.VISIBLE);
+                body.setText(message.getBody());
+            }
+
+            if (message.getTrackId() == null) {
+                tvIncomingTrack.setVisibility(View.GONE);
+                ivPlayIncoming.setVisibility(View.GONE);
+                ivIncomingTrack.setVisibility(View.GONE);
+            } else {
+                tvIncomingTrack.setVisibility(View.VISIBLE);
+                ivPlayIncoming.setVisibility(View.VISIBLE);
+                ivIncomingTrack.setVisibility(View.VISIBLE);
+                SpotifyApi spotifyApi = new SpotifyApi();
+                spotifyApi.setAccessToken(token);
+                SpotifyService service = spotifyApi.getService();
+                LoadChatTrackTask loadChatTrackTask = new LoadChatTrackTask(itemView, token, service, false);
+                loadChatTrackTask.execute(message);
+
+
+            }
+        }
+
+    }
+
+    public class OutgoingMessageViewHolder extends MessageViewHolder {
+        ImageView imageMe;
+        TextView body;
+        TextView tvOutgoingTrack;
+        ImageView ivOutgoingTrack;
+        ImageView ivPlayOutgoing;
+
+        public OutgoingMessageViewHolder(View itemView) {
+            super(itemView);
+            imageMe = itemView.findViewById(R.id.ivProfileMe);
+            body = itemView.findViewById(R.id.tvBody);
+            tvOutgoingTrack = itemView.findViewById(R.id.tvOutgoingTrack);
+            ivOutgoingTrack = itemView.findViewById(R.id.ivOutgoingTrack);
+            ivPlayOutgoing = itemView.findViewById(R.id.ivPlayOutgoing);
+        }
+
+        @Override
+        public void bindMessage(Message message) {
+            ParseUser user = message.getSender();
+                if (user.get("profilePic") != null) {
+                    ParseFile image = (ParseFile) user.get("profilePic");
+                    Glide.with(mContext).load(image.getUrl()).into(imageMe);
+                  }
+            if (message.getBody() == null) {
+                body.setVisibility(View.GONE);
+            } else {
+                body.setVisibility(View.VISIBLE);
+                body.setText(message.getBody());
+            }
+
+            if (message.getTrackId() == null) {
+                tvOutgoingTrack.setVisibility(View.GONE);
+                ivOutgoingTrack.setVisibility(View.GONE);
+                ivPlayOutgoing.setVisibility(View.GONE);
+            } else {
+                tvOutgoingTrack.setVisibility(View.VISIBLE);
+                ivOutgoingTrack.setVisibility(View.VISIBLE);
+                ivPlayOutgoing.setVisibility(View.VISIBLE);
+                SpotifyApi spotifyApi = new SpotifyApi();
+                spotifyApi.setAccessToken(token);
+                SpotifyService service = spotifyApi.getService();
+                LoadChatTrackTask loadChatTrackTask = new LoadChatTrackTask(itemView, token, service, true);
+                loadChatTrackTask.execute(message);
```

Comment:
I've seen this a couple times; consider refactoring these lines into a descriptively named util function that explains what these lines do at a high-level.